Nosto/nosto-magento2
# Bug/fix error on out of stock configurable product

This error was logged while browsing out-of-stock configurable products that have zero configurations (associated simple products).

## Description
**Error:**

> Noticed exception 'Nosto\Exception\NonBuildableProductException' with message 'SQLSTATE[42000]: Syntax error or access violation: 1064 You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near ')))) AND (stock_status_index.stock_status = 1)) AND (e.created_in <= '1644307...' at line 3, query was: SELECT e.*, stock_status_index.stock_status AS is_salable FROM catalog_product_entity AS e INNER JOIN catalog_product_website AS product_website ON product_website.product_id = e.entity_id AND product_website.website_id IN(3) INNER JOIN cataloginventory_stock_status AS stock_status_index ON e.entity_id = stock_status_index.product_id WHERE ((((e.entity_id = 'in') OR (()))) AND (stock_status_index.stock_status = 1)) AND (e.created_in <= '1644307200') AND (e.updated_in > '1644307200')' in vendor/nosto/module-nostotagging/Model/Product/Builder.php:312

In **Model/Product/Repository.php**:

`getSkuIds` function calls for associated products of the configurable product:
```
$this->configurableType->getChildrenIds($product->getId());
```

If `getSkuIds` returns empty array then `stockProvider->getInStockProductIds()` will be querying for empty list.

## How Has This Been Tested?

1. Switch to developer mode.

```
bin/magento deploy:mode:set developer
```

2. Add a configurable product with no/zero configurations (associated simple products).
OR, remove all the configurable from an existing configurable product.

3. Clear Cache.

```
bin/magento cache:flush
```

4. Open the configurable product link in the browser

5. Check the debug log (`var/log/debug.log`). The error mentioned above is logged in the `debug.log` file.
